### PR TITLE
feat(cdk/overlay): add disposeOnNavigation

### LINKED ISSUE #27545: feat(CdkConnectedOverlay): Add `cdkConnectedOverlayDisposeOnNavigation` as @Input for the directive
### Feature Description

It would be nice, if `cdkConnectedOverlayDisposeOnNavigation` could be added as @Input parameter for the CdkConnectedOverlay directive.
Adding it could also be a possibility to add other missing properties from the OverlayConfig.

In the following file `...cdk/overlay/overlay-directive.ts` a new input `@Input('cdkConnectedOverlayDisposeOnNavigation') open: boolean = false;` would have to be added and the `private _buildConfig(): OverlayConfig` method would need modifications to support the new input.

### Use Case

Using the `CdkConnectedOverlay` directive with `disposeOnNavigation`. The directive produces a significantly smaller overhead than the overlay.create() solution.

### PR DESCRIPTION
**Pull request by Naaajii**
add disposeOnNavigation to overlay directive

fixes #27545

### CODE REVIEW

File: src/cdk/overlay/overlay-directives.ts [hunk 1]
```diff
@@ -230,6 +231,15 @@ export class CdkConnectedOverlay implements OnDestroy, OnChanges {
     this._push = coerceBooleanProperty(value);
   }
 
+  /** Whether the overlay should be disposed of when the user goes backwards/forwards in history. */
+  @Input('cdkConnectedOverlayDisposeOnNavigation')
```
Comment (crisbeto) [inline src/cdk/overlay/overlay-directives.ts:235]: We're working towards using input transforms for this so let's start doing it here:

```
@Input({alias: 'cdkConnectedOverlayDisposeOnNavigation', transform: booleanAttribute})
```

Comment (Naaajii) [reply src/cdk/overlay/overlay-directives.ts]: I have only added transform here for `cdkConnectedOverlayDisposeOnNavigation` cause this PR is strictly for mentioned issue, should I send in other PRs with this Input transform refactoring? :grin:

Comment (crisbeto) [reply src/cdk/overlay/overlay-directives.ts]: Changing it in existing places will require a lot of debugging internally so it'll be easier if we handle it within the team.

**Review by crisbeto — APPROVED** [review]
LGTM

**Pull request merged**